rackcity/api/objects/rack_range.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework import serializers
+
+
+class RackRangeSerializer(serializers.Serializer):
+    letter_start = serializers.CharField(required=True, max_length=1)
+    letter_end = serializers.CharField(required=False, max_length=1)
+    num_start = serializers.IntegerField(required=True)
+    num_end = serializers.IntegerField(required=False)
+
+    def validate(self, data):
```

Comment:
The serializer `.is_valid()` method checks only that the JSON body has fields of the correct type and arguments (max_length, required). Added validation with this method to enforce that END fields are greater than START fields.

Reply:
I also thought about calling is_valid() within validate(), but I'm not sure that would be allowed since is_valid() is supposed to be the first thing called?  If that's allowed I think it would be cleaner

Reply:
Yeah I tried that too and it is not allowed (mentioned that in the other comment below)

`validate()` is a method on the `Serializer` class that I'm overriding. I just did some more investigating and this `validate()` method is the last thing that's called from `is_valid()`, so we don't need to do both here. The only bummer about that is only one validation exception is ever thrown from `is_valid()` so we couldn't specify which part of the request went wrong in the response. But I don't think that's a huge problem for now

Will update this PR with my changeszsszsszs

Reply:
Yeah I tried that too and it is not allowed (mentioned that in the other comment below)

`validate()` is a method on the `Serializer` class that I'm overriding. I just did some more investigating and this `validate()` method is the last thing that's called from `is_valid()`, so we don't need to do both here. The only bummer about that is only one validation exception is ever thrown from `is_valid()` so we couldn't specify which part of the request went wrong in the response. But I don't think that's a huge problem for now

Will update this PR with my changeszsszsszs

Reply:
Accidentally resolved these comments....I have pushed related updates!